### DISCUSSION TITLE
Cannot use custom audit on cached entities

### DISCUSSION THREAD

**Discussion by negreanucalin** [owen-it/laravel-auditing#787] [Q&A]
| Q                 | A
| ----------------- | ---
| Bug?              |yes
| New Feature?      | no
| Framework         | Laravel
| Framework version | 9.19
| Package version   | 13.0.5
| PHP version       | PHP 8.1.0 (cli) (built: Dec  2 2021 13:03:23) (NTS)

#### Actual Behaviour

Custom auditing using a cached version of the model does not insert any audits

#### Expected Behaviour

Audits should be inserted even if the model is cached

#### Steps to Reproduce

Behaviour:
Setup:
1. Set the `CACHE_DRIVER` to either `file` or `redis`

Steps:
1. Trigger custom event on entity (not cached), everything works fine
2. Second attempt (from cache) throws the event as well but the `RecordCustomAudit` listener doesn't catch it
for some reason
3. Clear cache
4. Repeat

Working sample:
```
$model->auditEvent = 'updated';
$model->isCustomEvent = true;
$model->auditCustomOld = [
    'demo' => [1,2]
];
$model->auditCustomNew = [
    'demo' => [2,4]
];
Event::dispatch(AuditCustom::class, [$model]);
```

Known:
1. With some logging found out that a custom listener catches the event (`AuditCustom::class`)

Unknown:
1. Why the default listener does not catch the event

#### Possible Solutions

N/A

**Comment by parallels999** [comment]
It was not designed to work with cached entities, but feel free to send a PR to add that functionality.
